api/testhelpers/proxy.go

```diff
@@ -86,9 +86,17 @@ func (p *ProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	go replicate(dconn, sconn)
 
 	// Wait until done, error, or 10 second.
-	select {
-	case <-time.After(10 * time.Second):
-	case <-errc:
+	numFinished := 0
+	for {
+		select {
+		case <-time.After(10 * time.Second):
+			return
+		case <-errc:
+			numFinished++
+			if numFinished == 2 {
+				return
+			}
+		}
 	}
```

Comment:
hmm I kinda worry that one of the replicate goroutines can just can stuck in a `io.Copy` call now, but maybe that's not the end of the world, there is a timeout.

But I do wonder, what is the scenario where one side of the connection has closed but we shouldn't close the other side? If that needs to be avoided, I think that would imply there's an issue with lib/utils.ProxyConn as well. Have you been able to reproduce the flake and have you seen improvement with this?

I am more suspicious of the second return from `hj.Hijack()` which we're currently ignoring, it's documented to be a bufio.Reader that may contain unprocessed buffered data from the client.

Reply:
Well, I thought I had reproduced it yesterday by delaying the responding handler, but I tried it again today and it's working fine, so 🤷

I also tried mocking a scenario where the `hj.Hijack()` buffer isn't empty, and that does cause an error. I have no idea if that's what's causing the flakiness (although if it is, there are several more places in Teleport where we're ignoring the hijack buffer that we should fix).